### DISCUSSION TITLE
How to generate a project from a local template with submodules?

### DISCUSSION THREAD

**Discussion by AbramovAV** [copier-org/copier#899] [Q&A] [answered]
I'm currently working on a template that has a submodule `converter` inside. When I try to render a project from this template (e.g. to check the correctness of generated project structure before pushing changes in template to the remote repo), I'm facing the following issue:
```
Submodule 'converter' (/template_path/converter.git) registered for path 'converter'
fatal: repository '/template_path/converter.git' does not exist
fatal: clone of '/template_path/converter.git' into submodule path '/tmp/copier.vcs.clone.4y5n3ykb/converter' failed
```
where `template_path` denotes a path to the local repository with template. The error itself appears after Copier tries to update all submodules with `git("submodule", "update", "--checkout", "--init", "--recursive", "--force")`.
It's important to mention that this issue appears only while rendering from local repo with template; everything works fine for the remote one.
As far as I know, this issue is mostly related to Git itself (which I tried to describe [here](https://stackoverflow.com/questions/74823785/update-submodules-for-a-git-repo-cloned-from-local-source)), but maybe there are some workarounds I am not familiar with?
I'm using Copier v7.0.1.

**Comment by yajo** [comment]
Could you please share the `template_path/.gitmodules` file?

**Reply by AbramovAV** [reply]
Sorry, couldn't attach the file itself as is, so just sharing all its contents:
```
[submodule "converter"]
	path = converter
	url = ../converter.git
```

Relative URL was used to clone this submodule inside of CI job in GitLab, as both submodule repo and the main project repo reside in the same group (like the documentation suggests to do [here](https://docs.gitlab.com/ee/ci/git_submodules.html#configure-the-gitmodules-file)).

**Comment by yajo** [comment, marked as answer]
Your will have to use absolute URL or path for the submodule.

**Reply by AbramovAV** [reply]
Thanks, that works perfectly!
